metrics/metrics.go

```diff
@@ -43,31 +47,30 @@ func (m *metric) SetTimestamp(t time.Time) {
 }
 
 // AddDimension will add this dimension with locking
-func (m *metric) AddDimension(key string, value interface{}) *metric {
+func (m *metric) AddDimension(key string, value interface{}) {
 	m.dimlock.Lock()
 	defer m.dimlock.Unlock()
```

Comment:
I guess you can also move this `Unlock` after the assignment and remove the `defer`.